packages/chevrotain/api.d.ts

```diff
@@ -866,7 +876,7 @@ export declare class Parser extends BaseParser {
         name: string,
         implementation: (...implArgs: any[]) => T,
         config?: IRuleConfig<T>
-    ): (idxInCallingRule?: number, ...args: any[]) => T | any
+    ): (idxInCallingRule?: number, ...args: any[]) => T
```

Comment:
this looks like a good fix.
It is not breaking change if I understand this correctly?